media/js/base/mozilla-smoothscroll.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+ Portions based on http://blog.gospodarets.com/native_smooth_scrolling/
+*/
+
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+Mozilla.smoothScroll = (function() {
+    'use strict';
+
+    var _smoothScroll;
+    var $htmlBody;
+
+    var _init = function(unitTest) {
+        var hasSmoothScroll;
+
+        if (unitTest) {
```

Comment:
Yeah, `unitTest` is dual purpose right now - forces re-initialization, and determines use of native scroll. Alternative could be passing in a different variable for testing to force re-init if you think that's better?